libdnf/advisory/advisory_query.cpp

```diff
@@ -301,22 +301,4 @@ std::vector<AdvisoryPackage> AdvisoryQuery::get_advisory_packages(
     return after_filter;
 }
 
-std::vector<AdvisoryPackage> AdvisoryQuery::get_sorted_advisory_packages(bool only_applicable) const {
-    std::vector<AdvisoryPackage> out;
-    for (Id candidate_id : *p_impl) {
-        Advisory advisory = Advisory(base, AdvisoryId(candidate_id));
-        auto collections = advisory.get_collections();
-        for (auto & collection : collections) {
-            if (only_applicable && !collection.is_applicable()) {
-                continue;
-            }
-            collection.get_packages(out);
-        }
-    }
-
-    std::sort(out.begin(), out.end(), AdvisoryPackage::Impl::nevra_compare_lower_id);
```

Comment:
But this might mean `AdvisoryQuery` could be dropped as `AdvisoryPackage`'s friend. We need to keep the friends lists as small as possible, it's already kind of crazy already...

Reply:
I am afraid `AdvisoryQuery` is still using `AdvisoryPackage::Impl::name_arch_compare_lower_id`.